realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -1095,11 +1095,8 @@ public void onChange(RealmResults<PermissionOffer> loadedOffers) {
         }
 
         void notifyCallbackWithSuccess(RealmResults<PermissionOffer> permissions) {
-            try {
-                callback.onSuccess(permissions);
-            } finally {
-                activeTasks.remove(this);
-            }
+            activeTasks.remove(this);
```

Comment:
It doesn't matter if the task is removed before or after really, so I opted for having to write less code :)